File: app/src/main/kotlin/org/triplepy/sh8email/sh8/activities/main/MainActivity.java
package org.triplepy.sh8email.sh8.activities.main;


import android.content.Context;
import android.os.Build;
import android.os.Bundle;
import android.support.annotation.Nullable;
import android.support.v7.widget.LinearLayoutManager;
import android.support.v7.widget.RecyclerView;
import android.transition.Transition;
import android.transition.TransitionInflater;

import org.triplepy.sh8email.sh8.R;
import org.triplepy.sh8email.sh8.activities.BaseActivity;

// FIXME: MAIN ACTIVITY!

public class MainActivity extends BaseActivity {
    private Context context;
    private RecyclerView recyclerView;

    @Override
    protected void onCreate(@Nullable Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        context = this;
        setupWindowAnimations();

        recyclerView = (RecyclerView) findViewById(R.id.recyclerView);
        recyclerView.setHasFixedSize(true);
        LinearLayoutManager linearLayoutManager = new LinearLayoutManager(context);
        recyclerView.setLayoutManager(linearLayoutManager);
        recyclerView.clearOnScrollListeners();
        // RecyclerView setting


    }

    private void setupWindowAnimations() {
        if (android.os.Build.VERSION.SDK_INT >= Build.VERSION_CODES.LOLLIPOP) {
            Transition fade = TransitionInflater.from(this).inflateTransition(R.transition.activity_fade);
            getWindow().setExitTransition(fade);
        }
    }

}

File: app/src/main/kotlin/org/triplepy/sh8email/sh8/activities/main/MainActivity.kt
package org.triplepy.sh8email.sh8.activities.main

import android.content.Context
import android.os.Bundle
import android.support.v7.widget.LinearLayoutManager
import android.transition.TransitionInflater
import kotlinx.android.synthetic.main.activity_main.*
import org.triplepy.sh8email.sh8.R
import org.triplepy.sh8email.sh8.activities.BaseActivity
import org.triplepy.sh8email.sh8.adapter.BaseAdapter


class MainActivity : BaseActivity() {

    private var context: Context? = null

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        context = this
        setContentView(R.layout.activity_main)
        setupWindowAnimations()

        // RecyclerView setting
        recyclerView.apply {
            setHasFixedSize(true)
            val linearLayout = LinearLayoutManager(context)
            layoutManager = linearLayout
            clearOnScrollListeners()

        }

        // Adapter setting
        initAdapter()
    }

    private fun initAdapter() {
        if (recyclerView.adapter == null) {
            recyclerView.adapter = BaseAdapter(this)
        }

    }


    private fun setupWindowAnimations() {
        val fade = TransitionInflater.from(this).inflateTransition(R.transition.activity_fade)
        window.exitTransition = fade
    }

}
